src/app/api/form/route.ts

```diff
@@ -0,0 +1,73 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "@database/db";
+import volunteerFormSchema, {
+  IVolunteerForm,
+  IFormQuestion,
+} from "@database/volunteerFormSchema";
+
+/**
+ * POST API for creating an VolunteerForm
+ * @returns None
+ */
+export async function POST(req: NextRequest) {
+  await connectDB();
+
+  try {
+    const { eventId, questions }: IVolunteerForm = await req.json();
+    if (eventId && Array.isArray(questions)) {
+      for (const question of questions) {
+        if (!question.question || !question.fieldType) {
+          console.error("Invalid request body - Missing required fields");
+          return NextResponse.json(
+            "Invalid request body - Missing required fields",
+            { status: 400 }
+          );
+        }
+        if (Array.isArray(question.options)) {
+          for (const option of question.options) {
+            if (typeof option != "string") {
+              console.error(
+                "Invalid request body - Incorrect type for options"
+              );
+              return NextResponse.json(
+                "Invalid request body - Incorrect type for options",
+                { status: 400 }
+              );
+            }
+          }
+        }
+      }
+    }
```

Comment:
Since the options array represents the options for a multiple choice question, it should be an error to try to create a question with type "MULTI_SELECT" or "MULTI_CHOICE" without an options array.

Reply:
I don't understand what you mean? So, are you saying that for the second if I need an else that follows and throws and error if it is not an array?

Reply:
If question.fieldType is "MULTI_SELECT" or "MULTI_CHOICE", then the options? field is no longer optional -- it's a required field. I think this logic should be implemented in your first if statement that checks for "invalid request body - missing required fields" errors.